BricksAndHearts.UnitTests/ServiceTests/Api/ApiServiceTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+
+namespace BricksAndHearts.UnitTests.ServiceTests.Api;
+
+public class ApiTests : ApiServiceTestsBase
+{
+    [Fact]
+    public async Task MakeApiRequestToAzureMaps_WhenCalled_ReturnsNonEmptyString()
+    {
+        // Arrange
+        string postalCode = "cb11dx";
+
+        // Act
+        // Uses the fake version of apiService because we don't want to make the API call
+        // Just want to check that the code works
+        string json = "some json string";
+        A.CallTo(() => _apiService.MakeApiRequestToAzureMaps(postalCode)).Returns(json);
+        string? responseBody = await _apiService.MakeApiRequestToAzureMaps(postalCode);
```

Comment:
This isn't testing anything! You're setting up and then calling a fake.

We need to fake `HttpClient` in some manner. We'll need to inject it for that, so changes ApiService to accept HttpClient as a parameter, and add `builder.Services.AddHttpClient();` to `Program.cs`.

Unfortunately the `GetAsync` method on `HttpClient` isn't virtual, so we can't fake it directly. There's two approaches:

* Fake `HttpMessageHandler` and pass it to `HttpClient`
* Wrap `HttpClient` with our own wrapper with virtual methods and then fake that

I've done the first here:

```csharp
// Setup
var options = A.Fake<IOptions<AzureMapsOptions>>();
var logger = A.Fake<ILogger<ApiService>>();
var fakeMessageHandler = A.Fake<HttpMessageHandler>();
var fakeHttpClient = new HttpClient(fakeMessageHandler);
var underTest = new ApiService(logger, options, fakeHttpClient); // Note new argument

// Arrange
const string postalCode = "cb11dx";
var responseBody = await File.ReadAllTextAsync($"{AppDomain.CurrentDomain.BaseDirectory}/../../../ServiceTests/Api/AzureMapsApiResponse.json");
var response = new HttpResponseMessage
{
    StatusCode = HttpStatusCode.OK,
    Content = new StringContent(responseBody)
};
// Slightly icky because "SendAsync" is protected
A.CallTo(fakeMessageHandler).Where(c => c.Method.Name == "SendAsync").WithReturnType<Task<HttpResponseMessage>>().Returns(response);

// Act
var result = await underTest.MakeApiRequestToAzureMaps(postalCode);

// Assert
result.Should().NotBeNullOrEmpty("should not be null or empty");
```

Reply:
yeah sure - I have read and copied your code. Will definitely do something similar next time I come across stuff like this!